src/pybind/mgr/dashboard/frontend/src/app/shared/services/notification.service.ts

```diff
@@ -63,17 +69,30 @@ export class NotificationService {
     localStorage.setItem(this.KEY, JSON.stringify(recent));
   }
 
+  queueNotifications(notifications: CdNotificationConfig[], ms: number = 500) {
+    this.queuedNotifications = this.queuedNotifications.concat(notifications);
+    window.setTimeout(() => {
+      this.sendQueuedNotifications();
+    }, ms);
+  }
+
+  private sendQueuedNotifications() {
+    this.utils.makeSet(this.queuedNotifications).forEach((notification) => {
+      this.show(notification);
+    });
+    this.queuedNotifications = [];
+  }
+
   /**
    * Method for showing a notification.
-   * @param {NotificationType} type toastr type
-   * @param {string} title
-   * @param {string} [message] The message to be displayed. Note, use this field
+   * @param {string} [config.message] The message to be displayed. Note, use this field
    *   for error notifications only.
-   * @param {*} [options] toastr compatible options, used when creating a toastr
-   * @memberof NotificationService
+   * @param {*} [config.options] toastr compatible options, used when creating a toastr
    * @returns The timeout ID that is set to be able to cancel the notification.
    */
-  show(type: NotificationType, title: string, message?: string, options?: any) {
+  show(config: CdNotificationConfig): number {
+    const { type, title, options } = config;
```

Comment:
Can't we keep the old parameters  and just instantiate the object here? That way we don't need to had overhead in all the other files.
```suggestion
      new CdNotificationConfig(type, title, message, options);
```

Reply:
I'll try to use it now as a overloaded function which can take both, in order to be free in use, object parameters, the classical parameters or the class object via it's constructor, which also takes the classic parameters.